### PR TITLE
SNOW-1421247: Remove skipped doctests that previously defaulted to pandas

### PR DESCRIPTION
**Pull request by sfc-gh-joshi**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1421247

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

Removes doctests that previously defaulted to pandas, but now fail.

These can be identified by their presence in the modin dependency PR in our old repo (https://github.com/snowflakedb/snowpandas/pull/995/files), or grepping for SNOW-1336091 and doctest: +SKIP.

I did not touch the docstrings for `apply` and `transform` even though they mostly fail. Most of these examples previously defaulted to pandas, so removing them would remove most of the docstring.

### CODE REVIEW

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py [hunk 1]
```diff
@@ -283,14 +282,6 @@ def dropna():
              name        toy       born
         1  Batman  Batmobile 1940-04-25
 
-        Drop the columns where at least one element is missing.
-
```
Comment (sfc-gh-vbudati) [inline src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:287]: Why does this not work anymore?

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py [hunk 2]
```diff
@@ -1269,15 +1250,6 @@ def fillna():
         2  0.0  1.0  2.0  3.0
         3  0.0  3.0  2.0  4.0
 
-        Only replace the first NaN element.
```
Comment (sfc-gh-vbudati) [inline src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:1272]: why doesn't this work as well?

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py [hunk 3]
```diff
@@ -2272,8 +2243,6 @@ def rename():
 
         >>> df.index
         Index([0, 1, 2], dtype='int64')
-        >>> df.rename(index=str).index  # doctest: +SKIP
```
Comment (sfc-gh-vbudati) [inline src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:2275]: I think this should work too

**Review by sfc-gh-oplaton — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py (hunk 1 above)
Comment (sfc-gh-joshi) [reply src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:287]: dropna with axis=1 isn't supported

https://github.com/snowflakedb/snowpark-python/blob/75cba1001fd7d09c40ea76a67a0320176cfc4942/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py#L7970

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py (hunk 2 above)
Comment (sfc-gh-joshi) [reply src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:1272]: fillna doesn't support limit

https://github.com/snowflakedb/snowpark-python/blob/75cba1001fd7d09c40ea76a67a0320176cfc4942/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py#L7856-L7859

File: src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py (hunk 3 above)
Comment (sfc-gh-joshi) [reply src/snowflake/snowpark/modin/plugin/docstrings/dataframe.py:2275]: `str` is a callable, and rename doesn't support callables

https://github.com/snowflakedb/snowpark-python/blob/75cba1001fd7d09c40ea76a67a0320176cfc4942/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py#L4625-L4629

**Review by sfc-gh-vbudati — APPROVED** [review]
Thanks Jonathan!

**Pull request merged**